README.md

```diff
@@ -26,11 +28,32 @@ Setup a logger with simple, unstructured logging and give it a try!
 
 [ECS] compliant JSON logging is also an option if we pick the correct reporter.
 ```ocaml
-# Logs.set_reporter (Ezlogs_cli.Json.reporter Fmt.stdout);;
+# Logs.set_reporter (Ezlogs_cli.Json_output.reporter Fmt.stdout);;
 - : unit = ()
 # Logs.info (fun m -> m "Hello from the future" ~tags);;
 {"@timestamp":"2052-02-05T20:58:14.386Z","log.level":"info","log.logger":"application","message":"Hello from the future"}
 - : unit = ()
 ```
 
+We can add fields to include more details.
+```ocaml
+# let error_tags = Ezlogs_cli.Ecs.add_tag (Error (Code "failure-101")) tags;;
+val error_tags : Logs.Tag.set = <abstr>
+# Logs.err (fun m -> m "There was a failure" ~tags:error_tags);;
+{"@timestamp":"2052-02-05T20:58:14.386Z","error.code":"failure-101","log.level":"error","log.logger":"application","message":"There was a failure"}
```

Comment:
I have some food for thought here:

  - The `log.logger` value _may be_ overridden by either Filebeat or Logstash
  - `error.type` - https://www.elastic.co/guide/en/ecs/current/ecs-error.html would be super beneficial as we can agg over it. But this might be tricky to implement?
  - There isn't a lot to aggregate over here. This will limit our dashboarding & log monitoring capabilities. I think we should just get rid of the message unless it is important and put in [events](https://www.elastic.co/guide/en/ecs/current/ecs-event.html). Without going off in a tangent, replacing `message` with `event.action` will allow the developer to quickly realize if their log delivers any value. And having a `file-created` event is much better than having 3 logs saying something like `opening file handler $filename`, `writing to file`, `closing file`. To me tags are useful, but they seem to be more of a global concept.

Reply:
- I was wondering about the circumstances under which `log.logger` (and honestly `log.*`) would be overwritten by Filebeat.  From what I've read I think we can setup the config appropriately to avoid having it overwritten but I haven't tried it yet.
- `error.type` is [in the code](https://github.com/hcarty/ezlogs-cli/pull/3/files#diff-5ab8778f2f1fd8509c6ec40348d088cbR50) now but it takes an exception as the payload and renders output from that.   I like that for simplicity and ease of use but I think we'll have to dogfood it a bit before we know if it's the right choice.

For aggregation, I'd see two things we'd potentially aggregate over in a simple log message like this:
1. `log.logger` because this will tell us the source of the logs (cohttp?  internal libraries? etc).
1. `error.code` (or `event.*` if we use the `event` hierarchy more generically) because this will let us group error/even cases.

I completely agree that the `event` fields need to be represented here.  I'll probably add those before merging.

For `message`, I think that's most useful to tell what is special about the log site.  If there isn't anything special then we could leave `message` blank or even filter it out of the JSON if `message = ""`.

Once this is merged, maybe you could create a PR with some README updates (or `*.ml` examples or tests) to see where the sharp edges are?  Or it's something we could pair on next week 💻